codegenerator/cli/templates/dynamic/codegen/src/IO.res.hbs

```diff
@@ -204,52 +210,70 @@ let executeBatch = async (sql, ~inMemoryStore: InMemoryStore.t, ~isInReorgThresh
 
 module RollBack = {
   exception DecodeError(S.error)
-  let rollBack = async (~chainId, ~blockTimestamp, ~blockNumber, ~logIndex) => {
-    let reorgData = switch await DbFunctions.sql->DbFunctions.EntityHistory.getRollbackDiff(
-      ~chainId,
-      ~blockTimestamp,
-      ~blockNumber,
-    ) {
-    | Ok(v) => v
-    | Error(exn) =>
-      exn
-      ->DecodeError
-      ->ErrorHandling.mkLogAndRaise(~msg="Failed to get rollback diff from entity history")
-    }
-
+  let rollBack = async (
+    ~chainId,
+    ~blockTimestamp,
+    ~blockNumber,
+    ~logIndex,
+    ~isUnorderedMultichainMode,
+  ) => {
     let rollBackEventIdentifier: Types.eventIdentifier = {
       chainId,
       blockTimestamp,
       blockNumber,
       logIndex,
     }
 
-    let inMemStore = InMemoryStore.makeWithRollBackEventIdentifier(Some(rollBackEventIdentifier))
-
-    //Don't save the rollback diffs to history table
-    let shouldSaveHistory = false
+    let inMemStore = InMemoryStore.make(~rollBackEventIdentifier)
 
-    reorgData->Belt.Array.forEach(e => {
-      switch e {
-      //Where previousEntity is Some, 
-      //set the value with the eventIdentifier that set that value initially
-      {{#each entities as | entity |}}
-      | {previousEntity: Some({entity: {{entity.name.capitalized}}(entity), eventIdentifier}), entityId} =>
-        inMemStore.{{entity.name.uncapitalized}}->InMemoryTable.Entity.set(
-          Set(entity)->Types.mkEntityUpdate(~eventIdentifier, ~entityId, ~shouldSaveHistory),
-          ~shouldSaveHistory,
-        )
-      {{/each}}
-      //Where previousEntity is None, 
-      //delete it with the eventIdentifier of the rollback event
-      {{#each entities as | entity |}}
-      | {previousEntity: None, entityType: {{entity.name.capitalized}}, entityId} =>
-        inMemStore.{{entity.name.uncapitalized}}->InMemoryTable.Entity.set(
-          Delete->Types.mkEntityUpdate(~eventIdentifier=rollBackEventIdentifier, ~entityId, ~shouldSaveHistory),
-          ~shouldSaveHistory,
+    await Utils.Array.awaitEach(Entities.allEntities, async entityMod => {
+      let module(Entity) = entityMod
+      let entityMod =
+        entityMod->(
+          Utils.magic: module(Entities.InternalEntity) => module(Entities.Entity with
+            type t = 'entity
+          )
         )
-      {{/each}}
-      }
+
+      let diff = await DbFunctions.sql->DbFunctions.EntityHistory.getRollbackDiff(
+        isUnorderedMultichainMode
+          ? UnorderedMultichain({
+              reorgChainId: chainId,
+              safeBlockNumber: blockNumber,
+            })
+          : OrderedMultichain({
+              safeBlockTimestamp: blockTimestamp,
+              reorgChainId: chainId,
+              safeBlockNumber: blockNumber,
+            }),
+        ~entityMod,
+      )
```

Comment:
The diffing now has a different way to treat ordered/unordered multichain mode there's a good explanation in DbFunctions.res